### ISSUE TITLE
no-duplicate-case incorrectly reports expressions

### ISSUE THREAD

**Issue by ZIJ** [eslint/eslint#2065]
The following code:
```javascript
switch(method) {
    case HttpMethod.Get:
        // handle GET
        break;
    case HttpMethod.Put:
        // handle PUT
        break;
}
```
reports "Duplicate case label" error. This is due to `getHash` implementation that's aware only about test identifier, but not the actual expression value:
```javascript
function getHash(test) {
        if (test.type === "Literal") {
            return test.type + typeof test.value + test.value;
        } else if (test.type === "Identifier") {
            return test.type + typeof test.name + test.name;
        }
    }
```

**Comment by lo1tuma** [comment]
Duplicate of #2038

**Comment by ZIJ** [comment]
@lo1tuma thanks for noticing! Also, that appears to be fixed in pull #2041. Closing.
